# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: parser/html/nsHtml5TreeBuilderCppSupplement.h
# Commit: 1954ce9669b8
# Full Hash: 1954ce9669b86b61c418c8286fbe0754ad2419cd
# Author: Adam Vandolder <avandolder@mozilla.com>
# Date: 2024-05-15 20:24:18
# Regressor Bug: 1890428
# File Overlap Count: 1
# Description:
#   Bug 1890428 - Add serialization support for Declarative Shadow DOM. r=webidl,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D206955
# ==============================================================================

diff -r 3e45e3d15eed -r 1954ce9669b8 parser/html/nsHtml5TreeBuilderCppSupplement.h
--- a/parser/html/nsHtml5TreeBuilderCppSupplement.h	Wed May 15 03:01:18 2024 +0000
+++ b/parser/html/nsHtml5TreeBuilderCppSupplement.h	Wed May 15 03:01:19 2024 +0000
@@ -1722,7 +1722,7 @@
   if (mBuilder) {
     nsIContent* root = nsContentUtils::AttachDeclarativeShadowRoot(
         static_cast<nsIContent*>(aHost), mode, aShadowRootIsClonable,
-        aShadowRootDelegatesFocus);
+        aShadowRootIsSerializable, aShadowRootDelegatesFocus);
     if (!root) {
       nsContentUtils::LogSimpleConsoleError(
           u"Failed to attach Declarative Shadow DOM."_ns, "DOM"_ns,
@@ -1738,9 +1738,9 @@
     return nullptr;
   }
   nsIContentHandle* fragHandle = AllocateContentHandle();
-  opGetShadowRootFromHost operation(aHost, fragHandle, aTemplateNode, mode,
-                                    aShadowRootIsClonable,
-                                    aShadowRootDelegatesFocus);
+  opGetShadowRootFromHost operation(
+      aHost, fragHandle, aTemplateNode, mode, aShadowRootIsClonable,
+      aShadowRootIsSerializable, aShadowRootDelegatesFocus);
   treeOp->Init(mozilla::AsVariant(operation));
   return fragHandle;
 }